src/Icon.js

```diff
@@ -70,7 +70,18 @@ const allKeys = Object.keys({
 })
 
 Icon.propTypes = {
-  name: PropTypes.oneOf(allKeys).isRequired,
+  name: props => {
+    const name = props.name
+    if (aliases[name] && !icons[name] && !icons.legacy[name]) {
+      console.warn(`Using '${name}' as an Icon name has been deprecated.`)
```

Comment:
It'd be nice to say *use <iconKey> instead* so that people know what the new name is